### PR TITLE
ci: address OSSF Scorecard warnings

### PR DESCRIPTION
**Pull request by alessfg**
### Proposed changes

This PR aims to:
* Remediate the `Token-Permissions` warning/low score by moving the F5 CLA GH Action permissions around.
* Remediate the `Pinned-Dependencies` warning/low score by using specific Docker image SHAs and adding Docker image updates to Dependabot.
* Hopefully remediate the `Vulnerabilities` warning/low score by recreating `package-lock.json`.
* Add both `workflow_dispatch` and `cron` builds to the OSSF Scorecard and S3 Gateway GitHub Actions in order to more proactively test changes (and to ensure that the pipelines run at minimum once a week).

### Checklist

Before creating a PR, run through this checklist and mark each as complete:

- [x] I have read the [contributing guidelines](/CONTRIBUTING.md).
- [x] I have signed the [F5 Contributor License Agreement (CLA)](https://github.com/f5/.github/blob/main/CLA/cla-markdown.md).
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works.
- [x] If applicable, I have checked that any relevant tests pass after adding my changes.
- [x] I have updated any relevant documentation (e.g. [`README.md`](/README.md)).

### CODE REVIEW

**Comment by alessfg** [comment]
recheck

**Comment by alessfg** [comment]
I have hereby read the F5 CLA and agree to its terms

**Comment by alessfg** [comment]
recheck

**Review by 4141done — CHANGES_REQUESTED** [review]
A few things here will break our builds but the other stuff looks good!

File: Dockerfile.latest-njs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # This container image removes the existing njs package from the inherited image
 # (which could be OSS NGINX or NGINX Plus), builds njs from the latest
 # source, and installs it.
-FROM nginx-s3-gateway
+FROM nginxinc/nginx-s3-gateway@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4
```
[the remark this answers is not in the document]
Comment (alessfg) [reply Dockerfile.latest-njs]: Fixed in the latest commit :)

File: Dockerfile.unprivileged [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # to allow running NGINX S3 Gateway as a non root user.
 # Steps are based on the official unprivileged container:
 # https://github.com/nginxinc/docker-nginx-unprivileged/blob/main/Dockerfile-debian.template
-FROM nginx-s3-gateway
+FROM nginxinc/nginx-s3-gateway@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4
```
[the remark this answers is not in the document]
Comment (alessfg) [reply Dockerfile.unprivileged]: Fixed in the latest commit :)

File: examples/brotli-compression/Dockerfile.oss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM nginxinc/nginx-s3-gateway
+FROM nginxinc/nginx-s3-gateway@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4
```
[the remark this answers is not in the document]
Comment (alessfg) [reply examples/brotli-compression/Dockerfile.oss]: The three examples here already pull the image from Docker Hub so including the latest available SHA should be ok?

File: examples/gzip-compression/Dockerfile.oss [hunk 4]
```diff
@@ -1,3 +1,3 @@
-FROM nginxinc/nginx-s3-gateway
+FROM nginxinc/nginx-s3-gateway@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4
```
[the remark this answers is not in the document]
Comment (alessfg) [reply examples/gzip-compression/Dockerfile.oss:1]: See previous comment

File: examples/modsecurity/Dockerfile.oss [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM nginxinc/nginx-s3-gateway
+FROM nginxinc/nginx-s3-gateway@sha256:8aa48324479b3653b5936183cc97f2ca1aa9078d229042f1bca357834bd906f4
```
[the remark this answers is not in the document]
Comment (alessfg) [reply examples/modsecurity/Dockerfile.oss:1]: See previous comment

File: examples/gzip-compression/Dockerfile.oss (hunk 4 above)
[the remark this answers is not in the document]
Comment (alessfg) [reply examples/gzip-compression/Dockerfile.oss:1]: Dependabot should hopefully take care of Docker SHA updates! 😁
